plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/alarms/AlarmManager.java

```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    /**
+     * Raise an alarm indicating that a system required by the active Plant
+     * Description was not found in the Service Registry.
+     *
+     * @param systemId   ID of a Plant Description system.
+     * @param systemName Name of a Plant Description system, or null.
+     * @param metadata   Metadata of a Plant Description system, or null.
+     */
+    public void raiseSystemNotRegistered(
+        final String systemId,
+        final String systemName,
+        final Map<String, String> metadata
+    ) {
+        Objects.requireNonNull(systemId, "Expected system ID");
+        raiseAlarm(systemId, systemName, metadata, AlarmCause.SYSTEM_NOT_REGISTERED);
+    }
+
+    /**
+     * Raise an alarm indicating that a system required by the active Plant
+     * Description cannot be uniquely identified in the Service Registry.
+     *
+     * @param systemId   ID of a Plant Description system.
+     * @param systemName Name of a Plant Description system, or null.
+     * @param metadata   Metadata of a Plant Description system, or null.
+     */
+    public void raiseMultipleMatches(
+        final String systemId,
+        final String systemName,
+        final Map<String, String> metadata
+    ) {
+        Objects.requireNonNull(systemId, "Expected system ID");
+        raiseAlarm(systemId, systemName, metadata, AlarmCause.MULTIPLE_MATCHES);
+    }
+
+    /**
+     * Raise an alarm indicating that a supposedly monitorable system does not
+     * seem to provide a 'Monitorable' service.
+     *
+     * @param systemId   ID of a Plant Description system.
+     * @param systemName Name of a Plant Description system, or null.
+     * @param metadata   Metadata of a Plant Description system, or null.
+     */
+    public void raiseSystemNotMonitorable(
+        final String systemId,
+        final String systemName,
+        final Map<String, String> metadata
+    ) {
+        Objects.requireNonNull(systemId, "Expected system ID");
+        raiseAlarm(systemId, systemName, metadata, AlarmCause.NOT_MONITORABLE);
+    }
+
+    /**
+     * Clear any alarms indicating that the system matching the given arguments
+     * does not provide a 'Monitorable' service.
+     *
+     * @param systemId   ID of a Plant Description system.
+     * @param systemName Name of a Plant Description system, or null.
+     * @param metadata   Metadata of a Plant Description system, or null.
+     */
+    public void clearSystemNotMonitorable(
+        final String systemId,
+        final String systemName,
+        final Map<String, String> metadata
+    ) {
+        Objects.requireNonNull(systemId, "Expected system ID");
+        final List<Alarm> cleared = activeAlarms.stream()
+            .filter(alarm ->
+                alarm.getCause() == AlarmCause.NOT_MONITORABLE &&
+                    Objects.equals(systemId, alarm.getSystemId()) &&
+                    Objects.equals(systemName, alarm.getSystemName()) &&
```

Comment:
maybe system name normalization can cause issues

Reply:
System names are now validated at the time they enter the PDE. This means that the system names used here are trimmed and on the correct format.